cmd/agent/api/agent/agent.go

```diff
@@ -381,6 +381,10 @@ func getFullRuntimeConfig(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if err := config.ResolveSecrets(config.Datadog, "datadog.yaml"); err != nil {
+		log.Warnf("Failed to resolve secrets: %s", err)
```

Comment:
Decided not to return after an error here since is unlikely that new secrets will be introduced in `datadog.yaml` where `additional_endpoints` are specified. An invalid secret in the main config file would have prevented the agent from starting anyways. Already known secrets are retrieved from cache. Worst thing this will return is the undecrypted form of the secret e.g. `ENC[foo]` which isn't sensitive.